service/history/visibilityQueueTaskExecutor.go

```diff
@@ -171,8 +171,7 @@ func (t *visibilityQueueTaskExecutor) processStartOrUpsertExecution(
 	}
 	startTimestamp := timestamp.TimeValue(startEvent.GetEventTime())
 	executionTimestamp := getWorkflowExecutionTime(mutableState, startEvent)
-	visibilityMemo := getWorkflowMemo(executionInfo.Memo)
-	// TODO (alex): remove copy?
```

Comment:
@alexshtin there is literally a comment here which suggests the opposite of what we are doing in this PR.  Do you have more context on why this comment exist?

Reply:
I think I didn't get the point of `release` call. Now it makes sense to me and I agree copy is required here. I would move funcs  to corresponding package not helper package/file. Like `copySearchAttributes` should go to `searchattribute` package.

Reply:
I believe it worth to leave comment next to `release` call saying that after this call no access to `mutableState` or `weContext` is allowed. Other devs might not get it too.